File: formal/coqimp/task05/extraction_hs/Fappli_IEEE_bits.hs
module Fappli_IEEE_bits where

import qualified Prelude
import qualified Fappli_IEEE
import qualified Fcore_Zaux

type Coq_binary64 = Fappli_IEEE.Coq_binary_float

default_nan_pl64 :: (,) Prelude.Bool Fappli_IEEE.Coq_nan_pl
default_nan_pl64 =
  (,) Prelude.False
    (Fcore_Zaux.iter_nat (\x -> (\x -> 2 Prelude.* x) x) (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ (Prelude.succ
      0))))))))))))))))))))))))))))))))))))))))))))))))))) 1)

unop_nan_pl64 :: Coq_binary64 -> (,) Prelude.Bool Fappli_IEEE.Coq_nan_pl
unop_nan_pl64 f =
  case f of {
   Fappli_IEEE.B754_nan s pl -> (,) s pl;
   _ -> default_nan_pl64}

b64_sqrt :: Fappli_IEEE.Coq_mode -> Fappli_IEEE.Coq_binary_float -> Fappli_IEEE.Coq_binary_float
b64_sqrt =
  Fappli_IEEE.coq_Bsqrt ((\x -> x) ((\x -> 2 Prelude.* x Prelude.+ 1) ((\x -> 2 Prelude.* x)
    ((\x -> 2 Prelude.* x Prelude.+ 1) ((\x -> 2 Prelude.* x) ((\x -> 2 Prelude.* x Prelude.+ 1)
    1)))))) ((\x -> x) ((\x -> 2 Prelude.* x) ((\x -> 2 Prelude.* x) ((\x -> 2 Prelude.* x)
    ((\x -> 2 Prelude.* x) ((\x -> 2 Prelude.* x) ((\x -> 2 Prelude.* x) ((\x -> 2 Prelude.* x)
    ((\x -> 2 Prelude.* x) ((\x -> 2 Prelude.* x) ((\x -> 2 Prelude.* x) 1))))))))))) unop_nan_pl64

File: formal/coqimp/task06/extraction_ml/Fappli_IEEE_bits.ml
open BinNums
open Datatypes
open Fappli_IEEE
open Fcore_Zaux

type binary64 = binary_float

(** val default_nan_pl64 : bool * nan_pl **)

let default_nan_pl64 =
  (false,
    (iter_nat (fun x -> Coq_xO x) (S (S (S (S (S (S (S (S (S (S (S (S (S (S
      (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S (S
      (S (S (S (S (S (S (S (S (S (S (S (S (S
      O))))))))))))))))))))))))))))))))))))))))))))))))))) Coq_xH))

(** val unop_nan_pl64 : binary64 -> bool * nan_pl **)

let unop_nan_pl64 = function
| B754_nan (s, pl) -> (s, pl)
| _ -> default_nan_pl64

(** val b64_sqrt : mode -> binary_float -> binary_float **)

let b64_sqrt =
  coq_Bsqrt (Zpos (Coq_xI (Coq_xO (Coq_xI (Coq_xO (Coq_xI Coq_xH)))))) (Zpos
    (Coq_xO (Coq_xO (Coq_xO (Coq_xO (Coq_xO (Coq_xO (Coq_xO (Coq_xO (Coq_xO
    (Coq_xO Coq_xH))))))))))) unop_nan_pl64
